scripts/release/atomic_pipeline.py

```diff
@@ -0,0 +1,552 @@
+#!/usr/bin/env python3
+
```

Comment:
how do we test this? Do we have a related unit test file integrated?

Reply:
The previous tests we had in `pipeline_test.py` were mostly testing some internals that are not used anymore i.e. `run_command_with_retries`, `is_version_in_range` or `test_is_release_step_executed` (sonar stuff) 

Apart from e2e tests that verify image creation I would also ask for:
- moving some tests that still qualify for `atomic_pipeline.py`:
  - `test_build_latest_agent_versions`
  - `test_get_versions_to_rebuild_same_version`
  - possibly others
- add tests for `process_image` and verify what commands are actually invoked using mocks?

Most complex building process is for `agent` image, but this will greatly change after non-matrix is merged, so I would wait with adding tests for it.

Reply:
I agree with what Maciej said, all unit tests were for low level internals. I don't think the pipeline is very relevant to test with unit tests.